.github/workflows/comment-on-asciidoc-changes.yml

```diff
@@ -11,19 +15,19 @@ jobs:
       - name: Checkout the repository
         uses: actions/checkout@v4
         with:
-          fetch-depth: 0  # This is important to fetch all history
+          # This is considered a security risk when used in conjunction with pull_request_target
+          # However, we are not running any code from the PR, so it's safe
+          ref: ${{ github.event.pull_request.head.sha }}
```

Comment:
Maybe add a link on why its a security risk: https://github.com/actions/checkout?tab=readme-ov-file#usage does not mention this :)

fetch-depth: 1 is suppose to work to according to the changed-files github actions docs?

Reply:
```suggestion
          # https://securitylab.github.com/resources/github-actions-preventing-pwn-requests/
          ref: ${{ github.event.pull_request.head.sha }}
```

Reply:
https://github.com/elastic/docs-builder/pull/367/commits/27d451849f54416e6f4b13df0e849352be71069d

Reply:
> fetch-depth: 1 is suppose to work to according to the changed-files github actions docs?

Yes, the examples in https://github.com/tj-actions/changed-files for `pull_request` always show the checkout action without any additional inputs.

And as stated in the usage, it's only necessary for the `push` event trigger.

Reply:
I was able to test it here: https://github.com/elastic/docs-eng-playground/pull/17